File: Accupay.Utils/TimeUtility.vb
Public Class TimeUtility

    Public Shared Function RangeStart(dateToday As Date, startTime As TimeSpan) As Date
        Return Combine(dateToday, startTime)
    End Function

    Public Shared Function RangeEnd(dateToday As Date, startTime As TimeSpan, endTime As TimeSpan) As Date
        Dim dateTomorrow = dateToday.AddDays(1)
        Return Combine(
            If(endTime > startTime, dateToday, dateTomorrow),
            endTime
        )
    End Function

    Public Shared Function Combine(day As Date, time As TimeSpan) As Date
        Dim timestampString = $"{day.ToString("yyyy-MM-dd")} {time.ToString("hh\:mm\:ss")}"

        Return Date.Parse(timestampString)
    End Function

    Public Shared Function ToDateTime(timeSpan As TimeSpan?) As DateTime?
        If Not timeSpan.HasValue Then
            Return Nothing
        End If

        Dim _today = Date.Now
        Dim _value = timeSpan.GetValueOrDefault

        Dim _dateTime =
            New Date(_today.Year, _today.Month, _today.Day,
                     _value.Hours, _value.Minutes, 0)

        Return _dateTime
    End Function

End Class

File: AccuPay.Utilities/TimeUtility.cs
using System;

namespace AccuPay.Utilities
{
    public static class TimeUtility
    {
        public static DateTime RangeStart(DateTime dateToday, TimeSpan startTime)
        {
            return Combine(dateToday, startTime);
        }

        public static DateTime RangeEnd(DateTime dateToday, TimeSpan startTime, TimeSpan endTime)
        {
            var dateTomorrow = dateToday.AddDays(1);
            return Combine(
                endTime > startTime ? dateToday : dateTomorrow,
                endTime
            );
        }

        public static DateTime Combine(DateTime day, TimeSpan time)
        {
            var timestampString = $"{day.ToString("yyyy-MM-dd")} {time.ToString(@"hh\:mm\:ss")}";

            return DateTime.Parse(timestampString);
        }

        public static DateTime? ToDateTime(TimeSpan? timeSpan, DateTime date)
        {
            if (!timeSpan.HasValue) return null;

            var dateTime = new DateTime(
                date.Year, date.Month, date.Day,
                timeSpan.Value.Hours, timeSpan.Value.Minutes, 0);

            return dateTime;
        }
    }
}